pool/pool_manager.gno

```diff
@@ -39,82 +141,56 @@ func CreatePool(
 	token0Path string,
 	token1Path string,
 	fee uint32,
-	_sqrtPriceX96 string, // uint256
+	_sqrtPriceX96 string,
 ) {
 	common.IsHalted()
 	en.MintAndDistributeGns()
 
-	if token0Path == token1Path {
-		panic(addDetailToError(
-			errDuplicateTokenInPool,
-			ufmt.Sprintf("pool_manager.gno__CreatePool() || expected token0Path(%s) != token1Path(%s)", token0Path, token1Path),
-		))
-	}
-
-	// wrap first
-	if token0Path == consts.GNOT {
-		token0Path = consts.WRAPPED_WUGNOT
-	} else if token1Path == consts.GNOT {
-		token1Path = consts.WRAPPED_WUGNOT
-	}
-
-	// then check if token0Path == token1Path
-	if token0Path == token1Path {
-		panic(addDetailToError(
-			errDuplicateTokenInPool,
-			ufmt.Sprintf("pool_manager.gno__CreatePool() || expected token0Path(%s) != token1Path(%s)", token0Path, token1Path),
-		))
-	}
-
-	sqrtPriceX96 := u256.MustFromDecimal(_sqrtPriceX96)
+	poolInfo := newPoolInput(token0Path, token1Path, fee, _sqrtPriceX96)
```

Comment:
poolInfo needs initialized after wrap/unwrap check

Reply:
maybe like? https://github.com/gnoswap-labs/gnoswap/pull/379/commits/eff5fa9445c03078ace3aeff0f6cd5802faef913

Reply:
Do we have to re-initialize? Can't we just initialize once?

Reply:
I think it wouldn’t have been necessary to add additional initialization before the modification since the value of `tokenPathXX` inside `poolInfo` (now `newPoolParams`) is directly updated its fields in the `wrap` function. Also, I didn't found relevant code from the original. 🫠

Could you clarify exactly what needs to be done? For now, I’ve added this line to update it uniformly.